Add NetworkPolicy for Prometheus

ozdanborne opened the pull request.

ozdanborne committed:
Add NetworkPolicy for Prometheus

mgoodness commented on stable/prometheus/templates/alertmanager-networkpolicy.yaml:
```diff
@@ -0,0 +1,20 @@
+{{- if .Values.networkPolicy.enabled }}
+apiVersion: {{ .Values.networkPolicy.apiVersion }}
```
Let's take this off users' shoulders and use Helm's `.Capabilities` object to set the API version accordingly.

mgoodness commented on stable/prometheus/templates/alertmanager-networkpolicy.yaml:
```diff
@@ -0,0 +1,20 @@
+apiVersion: {{ .Values.networkPolicy.apiVersion }}
+kind: NetworkPolicy
+metadata:
+  name: {{ template "prometheus.alertmanager.fullname" . }}
```
Add standard labels (`app`, `component`, `chart`, `release`, `heritage`).

mgoodness commented on stable/prometheus/templates/alertmanager-networkpolicy.yaml:
```diff
@@ -0,0 +1,20 @@
+      - podSelector:
+          matchLabels:
+            release: {{ .Release.Name }}
+            component: {{ .Values.server.name }}
```
Needs quotes around the value.

mgoodness commented on stable/prometheus/Chart.yaml:
```diff
@@ -1,5 +1,5 @@
 name: prometheus
-version: 3.0.2
+version: 3.0.3
```
We're adding functionality, so let's make this a minor (not patch) version bump.

mgoodness commented on stable/prometheus/templates/kube-state-metrics-networkpolicy.yaml:
```diff
@@ -0,0 +1,20 @@
+          component: {{ .Values.server.name }}
+  - ports:
+    - port: 8080
+{{- end }}
```
Comments from `alertmanager-networkpolicy.yaml` apply here.

mgoodness commented on stable/prometheus/templates/server-networkpolicy.yaml:
```diff
@@ -0,0 +1,15 @@
+  ingress:
+    - ports:
+      - port: 9090
+{{- end }}
```
Comments from `alertmanager-networkpolicy.yaml` apply here as well.

mgoodness reviewed (changes requested):
Looks pretty good, but I think we can eliminate the need to specify `apiVersion`. Don't forget to rebase!

mgoodness commented:
@k8s-bot ok to test

ozdanborne committed:
Code review

ozdanborne commented:
@mgoodness I've been trying to implement `.Capabilities` to automatically provide the correct networkPolicy apiVersion, but have been running into a lot of issues.

For starters, comparator functions (`ge`, `le`) do not accept strings. Only `int`. Which is quite confusing as [there's](https://github.com/kubernetes/charts/blob/3e43eb735c2da44565aea81bd65bb155fbaa2272/stable/traefik/templates/deployment.yaml#L17) [a](https://github.com/kubernetes/charts/blob/74dda6cdc563d27935daccb2d97336bc5625ae66/stable/kube2iam/templates/daemonset.yaml#L56) [few](https://github.com/kubernetes/charts/blob/f2e6b169cbfb1c5f131042f14fec9f0f8794bb8e/stable/spotify-docker-gc/templates/daemonset.yaml#L6) places where charts are comparing strings.

I can get past that syntax error by casting them to integers. But then when I try to deploy my chart, `.Capabilities.KubeVersion.Minor` and `.Major` are always nil. 

Any pointers on what's going on? I've tried using helm v2.4.2 and v2.5.1

Edit: Forgot to say thank you for taking a look and providing those suggestions :smile:

ozdanborne commented:
/retest

ozdanborne commented:
Thanks for helping me to realize I wasn't passing the current context to my template functions @mgoodness .

I think I've addressed all feedback at this point. Look good? feedback is in 2nd commit. I can squash when ready to merge

mgoodness commented:
/lgtm
